plutus-core/plutus-ir/src/PlutusIR/Contexts.hs

```diff
@@ -133,10 +133,12 @@ data SplitMatchContext tyname name uni fun a = SplitMatchContext
   }
 
 extractTyArgs :: AppContext tyname name uni fun a -> Maybe [Type tyname uni a]
-extractTyArgs = go []
-  where go acc (TypeAppContext ty _ ctx) = go (ty:acc) ctx
-        go _ (TermAppContext{})          = Nothing
-        go acc AppContextEnd             = Just acc
+extractTyArgs = go id
```

Comment:
I think I'd prefer to use `DList` explicitly, this is not very obvious to people who aren't familiar with it. Or just reverse at the end. Either works.

Reply:
Looks like [I am not the first one](https://github.com/IntersectMBO/plutus/blob/b97e8be50470065f37948f62dcaa7d67f71b9a39/plutus-core/plutus-ir/src/PlutusIR/Contexts.hs#L73) to use this idiom ;-)

Reply:
Yep, that person was wrong ;)

Reply:
I think all options are fine here, including the currently written one.